### ISSUE TITLE
Getting 403 Forbidden rather than 401 Unauthorized when initial authorization token expires after 60 minutes

### ISSUE THREAD

**Issue by jcambre** [guilhermesad/rspotify#142]
I'm having a hard time debugging an auth issue, and am wondering if I'm overlooking something small.

I'm using Rails 5 with Devise for authentication. Things work great for the first 60 minutes after a user initially authenticates through Spotify, but my RSpotify calls fail once that initial token expires. Based on what's mentioned in the README, I initialize users like this in my User.rb model:

```
   def self.from_omniauth(auth)
     where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
       spotify_user = RSpotify::User.new(auth)
       user.email = spotify_user.email
       user.spotify_hash = spotify_user.to_hash
     end
   end
```

And recover the RSpotify user through the spotify_hash in a controller, and save the hash after each use:
```
      rspotify_user = RSpotify::User.new(user.spotify_hash)
      @top_tracks = rspotify_user.top_tracks(limit: 50, offset: 0, time_range: 'short_term')
      user.spotify_hash = rspotify_user.to_hash
      user.save
```

After 60 minutes, I get a 403 Forbidden (rather than 401 Unauthorized) exception. Following the trace, it looks it never makes it to `rescue RestClient::Unauthorized => e` within `def self.oauth_send`(https://github.com/guilhermesad/rspotify/blob/master/lib/rspotify/user.rb#L56), so there's no call to the `refresh_token(user_id)` method.

Any idea where I'm going wrong? Thanks for your help, and sorry if I'm missing something obvious!

**Comment by adaam2** [comment]
@jcambre I also ran into this a couple of days ago - I don't think the automatic token refresh element of this works as it should. If I have some time I might look into it in the next couple of days. I think in the meantime, you can make use of `rescue_from` to catch the exception in your base controller, and then manually call the refresh token method, and persist the changes to your user.

**Comment by ORoberts221** [comment]
Think I'm having the same issue. After an hour all my requests return with 404 Invalid Username which I've raised an issue about #148. Just added your change @adaam2 so I'll see if that fixes it.

**Comment by guilhermesad** [comment]
Fixed by #165. Please use `2.1.1` version. LMK if you face any more issues on this.
